cylc/flow/scripts/cylc.py

```diff
@@ -362,74 +363,170 @@ def cli_version(long=False):
     sys.exit(0)
 
 
+@contextmanager
+def pycoverage(cmd_args):
+    """Capture code coverage if configured to do so.
+
+    This requires Cylc to be installed in editible mode
+    (i.e. `pip install -e`) in order to access the coverage configuration
+    file, etc.
+
+    $ pip install -e /cylc/working/directory
+
+    Set the CYLC_COVERAGE env var as appropriate before running tests
+
+    $ export CYLC_COVERAGE=1
+
+    Coverage files will be written out to the working copy irrespective
+    of where in the filesystem the `cylc` command was run.
+
+    $ cd /cylc/working/directory
+    $ coverage combine
+    $ coverage report
+
+    For remote tasks the coverage files will be written to the cylc
+    working directory on the remote so you will have to scp them back
+    to your local working directory before running coverage combine:
+
+    $ cd /cylc/working/directory
+    $ ssh remote-host cd /cylc/remote/working/directory && covrage combine
+    $ scp \
+    >     remote-host/cylc/remote/working/directory/.coverage \
+    >    .coverage.remote-host.12345.12345
+    $ coverage combine
+    $ coverage report
+
+    Environment Variables:
+        CYLC_COVERAGE:
+            '0'
+                Do nothing / run as normal.
+            '1'
+                Collect coverage data.
+            '2'
+                Collect coverage data and log every command for which
+                coverage data was successfully recorded to
+                a .coverage_commands_captured file in the Cylc
+                working directory.
+
+    """
+    cylc_coverage = os.environ.get('CYLC_COVERAGE')
+    if cylc_coverage not in ('1', '2'):
+        yield
+        return
+
+    # import here to avoid unnecessary imports when not running coverage
+    import cylc.flow
+    import coverage
+    from pathlib import Path
+
+    # the cylc working directory
+    cylc_wc = Path(cylc.flow.__file__).parent.parent.parent
```

Comment:
```suggestion
    cylc_wc = Path(cylc.flow.__file__).parents[2]
```
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parents

Reply:
Nice